MapboxCoreNavigation/OfflineDirections.swift

```diff
@@ -0,0 +1,194 @@
+import Foundation
+import MapboxDirections
+import MapboxNavigationNative
+
+public typealias OfflineDirectionsCompletionHandler = (_ numberOfTiles: UInt64) -> Void
+
+enum OfflineRoutingError: Error, LocalizedError {
+    case unexpectedRouteResult(String)
+    case corruptRouteData(String)
+    case responseError(String)
+    
+    public var localizedDescription: String {
+        switch self {
+        case .corruptRouteData(let value):
+            return value
+        case .unexpectedRouteResult(let value):
+            return value
+        case .responseError(let value):
+            return value
+        }
+    }
+    
+    var errorDescription: String? {
+        return localizedDescription
+    }
+}
+
+struct OfflineDirectionsConstants {
+    static let offlineSerialQueueLabel = Bundle.mapboxCoreNavigation.bundleIdentifier!.appending(".offline")
+    static let unpackSerialQueueLabel = Bundle.mapboxCoreNavigation.bundleIdentifier!.appending(".offline.unpack")
+    static let offlineSerialQueue = DispatchQueue(label: OfflineDirectionsConstants.offlineSerialQueueLabel)
+    static let unpackSerialQueue = DispatchQueue(label: OfflineDirectionsConstants.unpackSerialQueueLabel)
+}
+
+/**
+ Defines additional functionality similar to `Directions` with support for offline routing.
```

Comment:
How does this protocol differ from the OfflineDirectionsProtocol defined by MapboxDirections.swift? I think this could become a source of confusion among developers.

Reply:
I agree that it can cause confusion, however, `NavigatorDirectionsProtocol` depends on nav-native so it should be defined here, while `OfflineDirectionsProtocol` is a good fit for MapboxDirections since it's only adding additional directions related API calls. Do you have an idea how to simplify it? The protocol could be moved to MapboxDirections’s `OfflineDirectionsProtocol` but the implementation must reside in MapboxCoreNavigation.

Reply:
I think the confusion would stem from how this protocol is described as “similar to `Directions` with support for offline routing”. So if we need to keep it as a separate protocol, then can we frame it as a kind of NavigationService that has Directions functionality built in?